apps/studio/src/components/CoreTabs.vue

```diff
@@ -332,16 +332,22 @@
         tab.unsavedChanges = true
         this.addTab(tab)
       },
-      openTableProperties({ table }) {
+      async openTableProperties({ table }) {
         const t = new OpenTab('table-properties')
         t.tableName = table.name
         t.schemaName = table.schema
         t.title = table.name
-
-        const existing = this.tabItems.find((tab) => tab.matches(t))
-        if (existing) return this.$store.dispatch('tabs/setActive', existing)
-
-        this.addTab(t)
+        
+        try {
+          if (table.columns.length === 0) {
+            await this.$store.dispatch('updateTableColumns', table)
```

Comment:
I'm not sure we need to do this - doesnt table properties update the table columns on load? If not, that's the place to do this because this way won't work after closing the app and restoring tabs.